agimus_controller/mpc.py

```diff
@@ -1,380 +1,326 @@
-            )
-            joint_vel_constraint = crocoddyl.ConstraintModelResidual(
-                self.state,
-                joint_vel_residual,
-                np.array([-1e-6] * self.nv),
-                np.array([1e-6] * self.nv),
-            )
-            constraints.addConstraint(f"{joint_name}_velocity", joint_vel_constraint)
+from mim_robots.pybullet.env import BulletEnvWithGround
+import pinocchio as pin
+import mpc_utils
+import pin_utils
+from ocp import OCPPandaReachingColWithMultipleCol
+from wrapper_panda import PandaRobot
+
+np.set_printoptions(precision=4, linewidth=180)
+
+
+class MPC:
+    def __init__(
+        self,
+        robot_simulator: PandaRobot,
+        OCP: OCPPandaReachingColWithMultipleCol,
+        max_iter: int,
+        env: BulletEnvWithGround,
+        TARGET_POSE_1: pin.SE3,
+        TARGET_POSE_2: pin.SE3,
+        T_sim: float,
+    ):
+        """Create the MPC class used to solve the OCP problem.
+
+        Args:
+            robot_simulator (PandaRobot): Wrapper of pinocchio and pybullet robot.
+            OCP (OCPPandaReachingColWithMultipleCol): OCP solved in the MPC.
+            max_iter (int): Number max of iterations of the solver.
+            env (BulletEnvWithGround): Pybullet environement.
+            TARGET_POSE_1 (pin.SE3): First target of the robot.
+            TARGET_POSE_2 (pin.SE3): Second target of the robot.
+            T_sim (float): total time of the simulation.
         """
-        return crocoddyl.IntegratedActionModelEuler(
-            crocoddyl.DifferentialActionModelFreeFwdDynamics(
-                self.state, self.actuation, running_cost_model, constraints
-            ),
-            self.DT,
-        )
-
-    def get_goal_placement_residual(self):
-        """Return vector of tracking costs for each sub path."""
-
-        return self.get_placement_residual(self.x_plan[-1, : self.nq])
-
-    def get_placement_residual(self, q):
-        """Return placement residual to the last position of the sub path."""
-        target = self.robot.placement(q, self.nq).copy()
-        return crocoddyl.CostModelResidual(
-            self.state,
-            crocoddyl.ResidualModelFramePlacement(
-                self.state, self.robot.model.getFrameId(self.last_joint_name), target
-            ),
-        )
-
-    def get_velocity_residual(self, joint_name):
-        """Return velocity residual of desired joint."""
-        vref = pin.Motion.Zero()
-        return crocoddyl.CostModelResidual(
-            self.state,
-            crocoddyl.ResidualModelFrameVelocity(
-                self.state,
-                self.robot.model.getFrameId(joint_name),
-                vref,
-                pin.WORLD,
-            ),
-        )
 
-    def get_control_residual(self, uref):
-        return crocoddyl.CostModelResidual(
-            self.state, crocoddyl.ResidualModelControl(self.state, uref)
+        # Robot wrapper
+        self._robot_simulator = robot_simulator
```

Comment:
Do we need this inside the MPC class?
I would expect to have the debugging features outside of the MPC.

```python
my_visualizer = MyVisulaizer()
mpc = MPC()
my_data = {}

for i in range(1000):
  t = float(i) * 0.001
  my_data[t] =  mpc.solve(t)
  my_visualizer.visualize(my_data)
```